### PR TITLE
Performance: remove perf macro and reduce printed logs

### PR DESCRIPTION
**Pull request by btian1**
PERFORMANCE_COUNTERS_RUN_AVERAGE is not an independent macro,
it depend on: PERFORMANCE_COUNTERS, so it is duplicated,
remove this macro and use one macro to control this module,
also reduced unnecessary profiling log print.

Signed-off-by: Baofeng Tian <baofeng.tian@intel.com>

### CODE REVIEW

**Review by lyakh — COMMENTED** [review]
why not remove it from Kconfig too then?

File: app/perf_overlay.conf [hunk 1]
```diff
@@ -1,2 +1 @@
 CONFIG_PERFORMANCE_COUNTERS=y
-CONFIG_PERFORMANCE_COUNTERS_RUN_AVERAGE=y
```
Comment (lgirdwood) [inline app/perf_overlay.conf:2]: Why do we not want to know average ? @singalsu ?

**Review by lgirdwood — COMMENTED** [review]
Dont we need both max peak and average ?

File: app/perf_overlay.conf (hunk 1 above)
Comment (singalsu) [reply app/perf_overlay.conf:2]: Yes, please keep average included, it's important for power consumption estimates and such. We need to know both average and max.

Also min. would be valuable. With all min/max/average we get good view of how component uses processing capablity. Why not add it too?

Comment (btian1) [reply app/perf_overlay.conf:2]: We definitely need avg, this PR just want to remove the avg macro and use one macro to cover performance measurement, avg is still there, use two macro seems redundant, one is ok.

Comment (btian1) [reply app/perf_overlay.conf:2]: Please @mmaka1 also take a look.

Comment (lgirdwood) [reply app/perf_overlay.conf:2]: > We definitely need avg, this PR just want to remove the avg **macro** and use one macro to cover performance measurement, avg is still there, use two macro seems redundant, one is ok.

ah, got you  - can you reword the commit message to state we are removing a duplication.

**Review by lgirdwood — COMMENTED** [review]
I think we also need to remove the Kconfig too.

File: app/perf_overlay.conf (hunk 1 above)
Comment (btian1) [reply app/perf_overlay.conf:2]: > > We definitely need avg, this PR just want to remove the avg **macro** and use one macro to cover performance measurement, avg is still there, use two macro seems redundant, one is ok.
> 
> ah, got you - can you reword the commit message to state we are removing a duplication.

Done

**Comment by btian1** [comment]
> why not remove it from Kconfig too then?

Sorry, I also realized this, missed previously, now, removed, please review again.

File: src/audio/component.c [hunk 2]
```diff
@@ -251,7 +251,7 @@ int comp_copy(struct comp_dev *dev)
 		ret = dev->drv->ops.copy(dev);
 
 #if CONFIG_PERFORMANCE_COUNTERS
-		perf_cnt_stamp(&dev->pcd, comp_perf_info, dev);
+		perf_cnt_stamp(&dev->pcd, perf_trace_null, dev);
 		perf_cnt_average(&dev->pcd, comp_perf_avg_info, dev);
 #endif
```
Comment (lgirdwood) [inline src/audio/component.c:256]: We are missing some output if we set null for stamp i.e.
```c
#define comp_perf_info(pcd, comp_p)					\
	comp_info(comp_p, "perf comp_copy peak plat %u cpu %u",		\
		  (uint32_t)((pcd)->plat_delta_peak),			\
		  (uint32_t)((pcd)->cpu_delta_peak))

#define comp_perf_avg_info(pcd, comp_p)					\
	comp_info(comp_p, "perf comp_copy samples %u period %u cpu avg %u peak %u",\
		  (uint32_t)((comp_p)->frames),            \
		  (uint32_t)((comp_p)->period),			    \
		  (uint32_t)((pcd)->cpu_delta_sum),			\
		  (uint32_t)((pcd)->cpu_delta_peak))

```
We lose ```plat_delta_peak``` with this change ?

Can you provide the perf log output before and after this change.

**Review by lgirdwood — COMMENTED** [review]
We need to see the logs before and after this change to make sure we have not dropped data.

**Review by kv2019i — COMMENTED** [review]
Looks otherwise good but losing "plat_delta_peak" is a bit unexpected outcome.

**Comment by btian1** [comment]
> plat_delta_peak

it is there, due to it is same with cpu_delta_peak, once in future, there is a need to print out different plat_delta_peak, we can add it later.

thanks
Tim

**Comment by lgirdwood** [comment]
> it is there, due to it is same with cpu_delta_peak, and did not collect plat_delta_peak can reduce a little overhead.

oh, it should be different. IIUC the "CPU" results are based on CCOUNT DSP register and "plat" results are based on the always running audio timer. This could be a bug if they are always the same value. @singalsu have you seen this ?

**Comment by btian1** [comment]
> 

# CONFIG_TIMING_FUNCTIONS is not set

#ifdef __ZEPHYR__
	#ifdef CONFIG_TIMING_FUNCTIONS
		#define perf_cnt_get_cpu_ts arch_timing_counter_get
	#else
		#define perf_cnt_get_cpu_ts sof_cycle_get_64
	#endif	/* CONFIG_TIMING_FUNCTIONS */
#else
	#define perf_cnt_get_cpu_ts() timer_get_system(cpu_timer_get())
#endif	/* __ZEPHYR__ */

/** \brief Initializes timestamps with current timer values. */
#define perf_cnt_init(pcd) do {						\
		(pcd)->plat_ts = sof_cycle_get_64();			\
		(pcd)->cpu_ts = perf_cnt_get_cpu_ts();			\
	} while (0)

from above code, you can see that they are same

**Comment by lgirdwood** [comment]
> from above code, you can see that they are same

Can you share the before/after output from the FW logs (say 10 - 15 lines of each) - this will help me and other see the delta from this change.

**Comment by btian1** [comment]
Sure, Liam, please check below, I may pasted more for your better understanding:

Before change:

INFO perf comp_copy peak plat 241 cpu 242
INFO perf_cycle task 0x9e08a1b0, dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690> cpu avg 82 peak 763
INFO ll timer avg 4699, max 7422, overruns 0
INFO perf comp_copy samples 48 period 1000 cpu avg 421 peak 435
INFO perf comp_copy samples 48 period 1000 cpu avg 238 peak 242
INFO perf comp_copy samples 48 period 1000 cpu avg 803 peak 815
INFO perf comp_copy samples 48 period 1000 cpu avg 485 peak 507
INFO perf_cycle task 0xbe093180, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 2549 peak 3931
INFO perf_cycle task 0xbe093440, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 675 peak 1032
INFO perf comp_copy peak plat 435 cpu 434
INFO perf comp_copy peak plat 239 cpu 239
INFO perf comp_copy peak plat 811 cpu 811
INFO perf comp_copy peak plat 489 cpu 489
INFO perf comp_copy samples 48 period 1000 cpu avg 461 peak 472
INFO perf comp_copy peak plat 435 cpu 435
INFO perf comp_copy peak plat 495 cpu 494
INFO perf comp_copy peak plat 459 cpu 459
INFO perf comp_copy peak plat 241 cpu 241
INFO perf comp_copy peak plat 814 cpu 813
INFO perf comp_copy peak plat 506 cpu 506
INFO perf comp_copy peak plat 468 cpu 469
INFO perf comp_copy peak plat 471 cpu 472
INFO perf comp_copy peak plat 241 cpu 242
INFO perf comp_copy peak plat 815 cpu 815
INFO perf_cycle task 0x9e08a1b0, dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690> cpu avg 82 peak 763
INFO ll timer avg 4701, max 7408, overruns 0
INFO perf comp_copy samples 48 period 1000 cpu avg 421 peak 435
INFO perf comp_copy samples 48 period 1000 cpu avg 238 peak 242
INFO perf comp_copy samples 48 period 1000 cpu avg 803 peak 815
INFO perf comp_copy samples 48 period 1000 cpu avg 486 peak 506
INFO perf_cycle task 0xbe093180, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 2550 peak 3921
INFO perf_cycle task 0xbe093440, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 675 peak 1080
INFO perf comp_copy peak plat 434 cpu 435

you can see from above:
1. cpu and plat are same.
2. too much print caused log looks like not good and sequence may be disturbed due to too much 1ms logs.

After change:

INFO perf_cycle task 0xbe093180, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 2547 peak 3898
INFO perf_cycle task 0xbe093440, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 674 peak 1086
INFO perf comp_copy samples 48 period 1000 cpu avg 461 peak 472
INFO perf_cycle task 0x9e08a1b0, dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690> cpu avg 82 peak 766
INFO ll timer avg 4697, max 7380, overruns 0
INFO perf comp_copy samples 48 period 1000 cpu avg 420 peak 426
INFO perf comp_copy samples 48 period 1000 cpu avg 236 peak 246
INFO perf comp_copy samples 48 period 1000 cpu avg 808 peak 814
INFO perf comp_copy samples 48 period 1000 cpu avg 481 peak 496
INFO perf_cycle task 0xbe093180, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 2547 peak 3900
INFO perf_cycle task 0xbe093440, pipe-task <f11818eb-e92e-4082-82a3-dc54c604ebb3> cpu avg 674 peak 1088
INFO perf comp_copy samples 48 period 1000 cpu avg 461 peak 472
INFO perf_cycle task 0x9e08a1b0, dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690> cpu avg 82 peak 769
INFO ll timer avg 4697, max 7382, overruns 0
INFO perf comp_copy samples 48 period 1000 cpu avg 420 peak 426
INFO perf comp_copy samples 48 period 1000 cpu avg 236 peak 245
INFO perf comp_copy samples 48 period 1000 cpu avg 808 peak 816
INFO perf comp_copy samples 48 period 1000 cpu avg 481 peak 499

Only task and component logs, printed, it is per second, acceptable for trace capability, also have infornmation for average and peak cycle.

**Comment by lgirdwood** [comment]
> Only task and component logs, printed, it is per second, acceptable for trace capability, also have infornmation for average and peak cycle.

@singalsu good for you ?

**Review by singalsu — APPROVED** [review]

**Review by andrula-song — APPROVED** [review]

**Pull request merged**